### ISSUE TITLE
Click in right audio channel on simple media restart.

### ISSUE THREAD

**Issue by sergju** [google/ExoPlayer#9957]
Click in right audio channel on simple media restart.
Reproduced on some streams
[test_stream.zip](https://github.com/google/ExoPlayer/files/8031742/test_stream.zip)

I use this code.

 Remember url on start to restart later the same url with bit other paramenters

>  private MediaSource createMediaSource(String url, long startOffsetMs) {
>       Uri videoURI = Uri.parse(url);
>       Uri.Builder videoUriBuilder = videoURI.buildUpon();
>       videoURI = videoUriBuilder.build();
>       Log.i(TAG, "create media source for url:" + videoURI);
>       MediaItem mediaItem = new MediaItem.Builder().setUri(videoURI).build();
>       return mMediaSourceFactory.createMediaSource(mediaItem);
> }
> 
>   private void seekTest(int keyCode)
>   {
>       MediaSource mediaSource = createMediaSource(mUrl, 0);
>       player.setMediaSource(mediaSource, true);
>       player.prepare();
>   }
> 

- ExoPlayer version number: 2.16.1
- Android version: 7
- Android device: H96 Pro 4K

Also reproduced on other device with android 7

**Comment by ojw28** [comment]
To clarify, is this device specific. Does it reproduce only on H96 Pro 4K and one other device you have? Do you have devices on which the issue does not reproduce? I was unable to reproduce the problem locally on a couple of devices.

**Comment by sergju** [comment]
Yes, I reproduced on other device - hisilicon platform, but also android 7
It wasn't reproduce on other device with android 9
